parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -582,6 +582,13 @@ public ReadBuilder withNameMapping(NameMapping newNameMapping) {
           optionsBuilder = ParquetReadOptions.builder();
         }
 
+        if (filter != null &&
+                schema.getAliases() != null &&
+                ParquetFilters.isSupportedFilter(filter, schema, caseSensitive)) {
+          optionsBuilder.useRecordFilter(filterRecords);
```

Comment:
Seems like what we want here is `useColumnIndexFilter` and not `useRecordFilter` else `readNextFilteredRowGroup()` is equivalent to `readNextRowGroup()` https://github.com/apache/parquet-mr/blob/fe9ab5d22c602776f44b027ad3cc05609eddba54/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java#L956

Reply:
I found that `useColumnIndexFilter` is true by default

Reply:
Yes, we set it true by default. But eventually, it will rely on the filter we pushed here skip pages.